extension-qr-code/src/org/eclipse/birt/extension/qrcode/views/QRCodePageGeneratorFactory.java

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************************
+ * Copyright (c) 2007 Actuate Corporation.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ *
+ * Contributors:
+ *  Actuate Corporation  - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.birt.extension.qrcode.views;
+
+import org.eclipse.birt.report.designer.ui.views.IPageGenerator;
+import org.eclipse.core.runtime.IAdapterFactory;
+
+/**
+ * RotatedTextPageGeneratorFactory
+ */
+public class QRCodePageGeneratorFactory implements IAdapterFactory {
+
+	public Object getAdapter(Object adaptableObject, Class adapterType) {
+		return new QRCodePageGenerator();
```

Comment:
It should return this only in case if `org.eclipse.birt.report.designer.ui.views.IPageGenerator` was requested

Reply:
Can you be more precise? I don't know what I should test here (the interface is too generic)

Reply:
Perhaps you are not familiar with this contract of `IAdaptapleFactory`
It is all about adapting `adaptableObject` to a type `adapterType`

So, what I would expect to see is like this
```
if(IPageGenerator.class.isInstance(adapterType)) {
    return new QRCodePageGenerator();   
}
return null;
```

Reply:
Added this test.